### PR TITLE
[torchlib] Fix aten_mean_dim

### PR DESCRIPTION
**Pull request by justinchuby**
Fix when the rank of `dim` is not known, the conditional will pick the wrong branch because the truthiness of a value is always True.

### CODE REVIEW

**Comment by justinchuby** [comment]
Validated with 

```py
import torch
import transformers

config = transformers.MistralConfig(
    hidden_size=32,
    num_hidden_layers=2,
    vocab_size=99,
    intermediate_size=16,
    max_position_embeddings=512,
    num_attention_heads=2,
    num_key_value_heads=2,
    sliding_window=4096,
)
config._attn_implementation = "eager"
model = transformers.MistralModel(config)
shape = (1, 30)
input_ids = torch.randint(0, 99, shape).to(torch.int64)
attention_mask = torch.ones(shape)
expected = model(input_ids, attention_mask)
ep = torch.export.export(model, (input_ids, attention_mask))
mod = ep.module()
got = mod(input_ids, attention_mask)

onnx_program = torch.onnx.export(ep)
onnx_program.save("mistral.onnx")
import torch.onnx._internal.exporter._testing as onnx_testing
onnx_testing.assert_onnx_program(onnx_program)
```

**Review by titaiwangms — APPROVED** [review]
